src/NCalc/LambdaExpressionVistor.cs

```diff
@@ -200,26 +230,76 @@ public override void Visit(Function function)
                 return;
             }
 
-            switch (functionName)
+            Action<string, int, int> CheckArgumentsLengthForFunction = delegate (string funcStr, int argsNum, int argsNeed)
             {
-                case "MIN":
-                    var minArg0 = L.Expression.Convert(args[0], typeof(double));
-                    var minArg1 = L.Expression.Convert(args[1], typeof(double));
-                    _result = L.Expression.Condition(L.Expression.LessThan(minArg0, minArg1), minArg0, minArg1);
-                    break;
-                case "MAX":
-                    var maxArg0 = L.Expression.Convert(args[0], typeof(double));
-                    var maxArg1 = L.Expression.Convert(args[1], typeof(double));
-                    _result = L.Expression.Condition(L.Expression.GreaterThan(maxArg0, maxArg1), maxArg0, maxArg1);
-                    break;
-                case "POW":
-                    var powArg0 = L.Expression.Convert(args[0], typeof(double));
-                    var powArg1 = L.Expression.Convert(args[1], typeof(double));
-                    _result = L.Expression.Power(powArg0, powArg1);
-                    break;
-                default:
-                    throw new MissingMethodException($"method not found: {functionName}");
+                if (argsNum != argsNeed)
+                    throw new ArgumentException($"{funcStr} takes exactly {argsNeed} argument");
+            };
+
+            Action<string, int> MakeMathCallExpressionOneDoubleParam = delegate (string mathCallFuncName, int argsNumActual)
+            {
+                CheckArgumentsLengthForFunction(mathCallFuncName, argsNumActual, 1);
+                var arg = L.Expression.Convert(args[0], typeof(double));
+                _result = L.Expression.Call(typeof(Math).GetMethod(mathCallFuncName), arg);
+            };
+
+            Action<string, int> MakeMathCallExpressionTwoDoubleParam = delegate (string mathCallFuncName, int argsNumActual)

```

Comment:
I wanted anonymous function inside method with small reusable functionality

Reply:
I think what you want is a [local function](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions), though I doubt you can use them here due to c# version requirements.  

There's overhead to declaring delegates like this.  Every time Visit is called you're declaring these delegates as a local variable.  A private static class method would perform much better. 

I wrote up a quick test to illustrate.

![image](https://github.com/sklose/NCalc2/assets/43828739/fad44ed7-459f-41d7-853f-901bda5f560f)
```
BenchmarkRunner.Run<TestInvoke>();

public class TestInvoke
{
    [Benchmark(Baseline = true)]
    public void InvokeDelegate()
    { 
        Action<string, int> MakeMathCallExpressionTwoDoubleParam = delegate (string mathCallFuncName, int argsNumActual)
        {
            if (argsNumActual > 1000)
            {
                throw new ArgumentOutOfRangeException();
            }
        };

        MakeMathCallExpressionTwoDoubleParam("one", 1);
    }

    [Benchmark]
    public void InvokeMethod()
    {
        MakeMathCallExpressionTwoDoubleParam("one", 1);
    }

    private static void MakeMathCallExpressionTwoDoubleParam(string mathCallFuncName, int argsNumActual)
    {
        if (argsNumActual > 1000)
        {
            throw new ArgumentOutOfRangeException();
        }
    }
}

```

Reply:
Thanks! Local methods are from C#7 so according to this
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version
It may have support

Reply:
But I found that .NET Framework 4.5 use C# 5

Reply:
If I could successfully create nupkg does it mean that I could use C# 7?

Reply:
I don't know the answer.  The compiler could simply remove all traces of the local function via inlining.